src/trezor.rs

```diff
@@ -0,0 +1,185 @@
+}
+
+impl std::fmt::Debug for TrezorClient {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TrezorClient")
+            .field("client", &self.client.borrow().model())
+            .finish()
+    }
+}
+
+impl TrezorClient {
+    fn new(client: Trezor) -> Self {
+        let kind = match client.model() {
+            trezor_client::Model::TrezorEmulator => DeviceKind::TrezorSimulator,
+            _ => DeviceKind::Trezor,
+        };
+        Self {
+            client: RefCell::new(client),
+            kind,
+            network: bitcoin::Network::Testnet,
+        }
+    }
+
+    pub fn connect_first(debug: bool) -> Result<Self, Error> {
+        let mut devices = trezor_client::find_devices(debug);
+        if !devices.is_empty() {
+            let mut client = devices.remove(0).connect()?;
+            client.init_device(None)?;
+            Ok(Self::new(client))
+        } else {
+            Err(Error::DeviceNotFound)
+        }
+    }
+}
+
+#[async_trait]
+impl HWI for TrezorClient {
+    fn device_kind(&self) -> crate::DeviceKind {
+        self.kind
+    }
+
+    async fn get_version(&self) -> Result<super::Version, Error> {
+        let client = self.client.borrow();
+        let f = client.features();
+        if let Some(f) = f {
+            let version = super::Version {
+                major: f.major_version(),
+                minor: f.minor_version(),
+                patch: f.patch_version(),
+                prerelease: None,
+            };
+            Ok(version)
+        } else {
+            return Err(Error::Device(String::from("No features found")));
+        }
+    }
+
+    async fn is_connected(&self) -> Result<(), Error> {
+        match self.client.borrow_mut().ping("PINGPING")? {
+            TrezorResponse::Ok(_) => Ok(()),
+            _ => Err(Error::DeviceDisconnected),
+        }
+    }
+    async fn get_master_fingerprint(&self) -> Result<Fingerprint, Error> {
+        let path = DerivationPath::default();
+        match self.client.borrow_mut().get_public_key(
+            &path,
+            trezor_client::InputScriptType::SPENDADDRESS,
+            bitcoin::Network::Bitcoin,
+            false,
+        ) {
+            Ok(TrezorResponse::Ok(key)) => {
+                let fp = key.fingerprint();
+                Ok(fp)
+            }
+            Ok(TrezorResponse::Failure(f)) => Err(Error::Device(f.to_string())),
+            Ok(result) => Err(Error::Device(result.to_string())),
+            Err(e) => Err(Error::Device(e.to_string())),
+        }
+    }
+
+    async fn get_extended_pubkey(&self, path: &DerivationPath) -> Result<ExtendedPubKey, Error> {
+        let path = DerivationPath::from_str(&path.to_string())
+            .map_err(|e| Error::Device(format!("{:#?}", e)))?;
+        match self.client.borrow_mut().get_public_key(
+            &path,
+            trezor_client::InputScriptType::SPENDADDRESS,
+            bitcoin::Network::Bitcoin,
```

Comment:
I think it should be
```suggestion
            self.network,
```
right ?

Reply:
Yep, sorry about that.